### PR TITLE
[Fix](tvf) Fix that tvf reading empty files in compressed formats.

### PR DESCRIPTION
**Pull request by BePPPower**
## Proposed changes

Issue Number: close #xxx

1. Fix the issue with tvf reading empty compressed files.

2. move two test cases (`test_local_tvf_compression` and `test_s3_tvf_compression`) from p2 to p0

docs: https://github.com/apache/doris-website/pull/654

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run p0

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/ExternalFileTableValuedFunction.java [hunk 1]
```diff
@@ -484,7 +484,8 @@ private PFetchTableSchemaRequest getFetchTableStructureRequest() throws TExcepti
         // get first file, used to parse table schema
         TBrokerFileStatus firstFile = null;
         for (TBrokerFileStatus fileStatus : fileStatuses) {
-            if (fileStatus.isIsDir() || fileStatus.size == 0) {
+            if (fileStatus.isIsDir() || fileStatus.size == 0
+                    || (compressionType != TFileCompressType.UNKNOWN && fileStatus.size <= 4)) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/ExternalFileTableValuedFunction.java:488]: Are you sure all compressed file size have at least 4 bytes?

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run p0

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by zddr — APPROVED** [review]

**Pull request merged**